examples/core-api.js

```diff
@@ -1,19 +1,41 @@
 'use strict';
 const os = require('os');
+const http = require('http');
 const libui = require('..');
 
 const win = new libui.UiWindow('Test window', 800, 600, false);
 win.margined = 1;
-win.onClosing(() => {
-	libui.stopLoop();
-});
 
 const box = new libui.UiVerticalBox();
 const hBox = new libui.UiHorizontalBox();
 const e1 = new libui.UiEntry();
 e1.enabled = 0;
-hBox.append(new libui.UiLabel('ciao'), false);
+const lblCiao = new libui.UiLabel('ciao');
+hBox.append(lblCiao, false);
+
 hBox.append(e1, false);
+let idxLbl = 0;
+
+const interval = setInterval(() => {
+	lblCiao.text = String(idxLbl++);
+}, 1000);
+
+// Create an HTTP tunneling proxy
+const proxy = http.createServer((req, res) => {
+	lblCiao.text = String(idxLbl++);
+	res.writeHead(200, {'Content-Type': 'text/plain'});
+	res.end(String(idxLbl));
+});
+proxy.listen(3000, '127.0.0.1', () => {
+	console.log('listening...');
+});
+
+win.onClosing(() => {
+	win.close();
+	clearInterval(interval);
+	proxy.close();
+	libui.stopLoop();
+});
 
 box.append(new libui.UiEntry(), false);
 box.append(hBox, false);
```

Comment:
Uhm I should revert changes to this, they was meant just to test while developing...

Reply:
Are you aware that "Promise", "NextTick" and "Http" only print something after manually clicking "setImmediate" ?